src/Sulu/Component/Util/Tests/Unit/SuluNodeHelperTest.php

```diff
@@ -254,19 +254,20 @@ public function testSiblingNodes(): void
                 ->method('getPath')
                 ->will($this->returnValue('/foobar/foobar-' . $i));
         }
-
+        $iterator = new \ArrayIterator([
+            $node1, $node2, $node3,
+        ]);
         $node2->expects($this->any())
             ->method('getParent')
             ->will($this->returnValue($this->node));
         $this->node->expects($this->any())
             ->method('getNodes')
-            ->will($this->returnValue([
-                $node1, $node2, $node3,
-            ]));
+            ->will($this->returnValue($iterator));
 
         $res = $this->helper->getNextNode($node2);
         $this->assertSame($node3->getPath(), $res->getPath());
 
+        $iterator->rewind();
```

Comment:
We need to rewind the iterators in between usages as Phpunit will not create the iterator again when the `getNodes` method  is called a second time.